pkg/controller/queuejob/queuejob_controller_ex.go

```diff
@@ -1665,23 +1671,30 @@ func (cc *XController) addQueueJob(obj interface{}) {
 		go func() {
 			for {
 				time.Sleep(requeueInterval)
-				latestAw, exists, err := cc.appwrapperInformer.Informer().GetStore().GetByKey(key)
+				latestObj, exists, err := cc.appwrapperInformer.Informer().GetStore().GetByKey(key)
 				if err != nil && !exists {
 					klog.Warningf("[Informer-addQJ] Recent copy of AW %s not found in cache", qj.Name)
 				} else {
-					if latestAw.(*arbv1.AppWrapper).Status.State != arbv1.AppWrapperStateActive && latestAw.(*arbv1.AppWrapper).Status.State != arbv1.AppWrapperStateEnqueued && latestAw.(*arbv1.AppWrapper).Status.State != arbv1.AppWrapperStateRunningHoldCompletion {
-						klog.V(2).Infof("[Informer-addQJ] Stopping requeue for AW %s with status %s", latestAw.(*arbv1.AppWrapper).Name, latestAw.(*arbv1.AppWrapper).Status.State)
+					var latestAw *arbv1.AppWrapper
+					if latestObj != nil {
+						latestAw = latestObj.(*arbv1.AppWrapper)
+					} else {
+						latestAw = qj
```

Comment:
I wonder what happens here to `qj` when we leave the scope of this function. This seems really dangerous. This `for` loop is going to keep retrying `qj` outside of `addQueueJob` since it is a thread. Does it retain a copy of `qj`? I'm not saying we should change it, I'm just concerned.